src/Functions/map.cpp

```diff
@@ -144,6 +146,97 @@ class FunctionMap : public IFunction
     }
 };
 
+// mapFromArrays(keys, values) is a function that allows you to make key-value pair from a pair of arrays
+class FunctionMapFromArrays : public IFunction
+{
+public:
+    static constexpr auto name = "mapFromArrays";
+
+    static FunctionPtr create(ContextPtr)
+    {
+        return std::make_shared<FunctionMapFromArrays>();
+    }
+
+    String getName() const override
+    {
+        return name;
+    }
+
+    size_t getNumberOfArguments() const override
+    {
+        return 2;
+    }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+    bool useDefaultImplementationForNulls() const override { return false; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments.size() != 2)
+            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                "Function {} requires 2 arguments, but {} given", getName(), arguments.size());
+
+        const auto * keys_type = checkAndGetDataType<DataTypeArray>(arguments[0].get());
+        if (!keys_type)
+            throw Exception{"First argument for function " + getName() + " must be a Array",
+                            ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+
+        const auto * values_type = checkAndGetDataType<DataTypeArray>(arguments[1].get());
+        if (!values_type)
+            throw Exception{"Second argument for function " + getName() + " must be a Array",
+                            ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+
+        DataTypes tmp;
+        const auto & key_type = keys_type->getNestedType();
+        const auto & value_type = values_type->getNestedType();
+        tmp.emplace_back(key_type);
+        tmp.emplace_back(value_type);
+        return std::make_shared<DataTypeMap>(tmp);
+    }
+
+    ColumnPtr executeImpl(
+        const ColumnsWithTypeAndName & arguments, const DataTypePtr & /* result_type */, size_t /* input_rows_count */) const override
+    {
+        bool is_keys_const = isColumnConst(*arguments[0].column);
+        const ColumnArray * col_keys;
+        if (is_keys_const)
+        {
+            auto tmp = arguments[0].column->convertToFullColumnIfConst();
+            col_keys = checkAndGetColumn<ColumnArray>(tmp.get());
```

Comment:
`tmp` is destroyed in this scope, are pointer `col_keys` valid outside ?

Reply:
@vdimir This idea is from spark udf: `map_from_arrays`, and I think it make tranforming from array to map more easier.

Reply:
> `tmp` is destroyed in this scope, are pointer `col_keys` valid outside ?

Fixed.